src/main/java/lai/util/Ui.java

```diff
@@ -0,0 +1,142 @@
+package lai.util;
+
+import lai.task.Task;
+
+import java.util.Scanner;
+
+/**
+ * Handles the user interface for the Lai application.
+ * This class is responsible for printing messages to the console
+ * and reading user input.
+ */
+public class Ui {
+    protected Scanner scanner;
+
+    /**
+     * Constructs a UI object with a given Scanner.
+     *
+     * @param scanner The scanner to read user input.
+     */
+    public Ui (Scanner scanner) {
+        this.scanner = scanner;
+    }
+
+    /**
+     * Prints a welcome message at the start of the application.
+     */
+    public static void printStart() {
+        System.out.println("Hi there, I am Lai. Your friendly fairly useless chatbot.");
+        System.out.println("What can I assist you with today?");
+        printLine();
+    }
+
+    /**
+     * Prints a farewell message when exiting the application.
+     */
+    public static void printBye() {
+        printLine();
+        System.out.println("Goodbye, we shall meet again. Hopefully.");
+        printLine();
+    }
+
+    /**
+     * Prints a decorative line to separate sections of output for better readability.
+     */
+    public static void printLine() {
```

Comment:
Consider "printHorizontalLine" or "printDecorativeLine". Since println in fact stands for printline.